From 03123b10b70dab80e9777181dcb519505bff3768 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 2 Dec 2016 11:38:58 +0100
Subject: [PATCH] Honor HTTPS_PROXY and https_proxy if env_proxy not specified
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Changing default SSL implementation from Net::SSL to IO::Socket::SSL
broke Net::SSL way for specifying HTTPS proxy via environment
variables.

This patch enables processing proxy environment variables if
LWP::UserAgent->new() env_proxy argument is not specified and
HTTPS_PROXY or https_proxy variable exists.

This patch also accepts proxy specification without schema to mimic
Net:SSL better.

<https://bugzilla.redhat.com/show_bug.cgi?id=1400632>

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 lib/LWP/UserAgent.pm | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/lib/LWP/UserAgent.pm b/lib/LWP/UserAgent.pm
index 6d526a4..034df50 100644
--- a/lib/LWP/UserAgent.pm
+++ b/lib/LWP/UserAgent.pm
@@ -85,6 +85,21 @@ sub new
     my $max_redirect = delete $cnf{max_redirect};
     $max_redirect = 7 unless defined $max_redirect;
     my $env_proxy = delete $cnf{env_proxy};
+    unless (defined $env_proxy) {
+        # The processing of proxy environment variables is for compatiblity
+        # with Net::SSL
+        my $value;
+        for my $name ('HTTPS_PROXY', 'https_proxy') {
+            next unless exists $ENV{$name};
+            $value = $ENV{$name};
+            if ($value !~ s|\A\w+://||i) {
+                $ENV{$name} = 'http://' . $value;
+            }
+        }
+        if (defined $value) {
+            $env_proxy = 1;
+        }
+    }
 
     my $cookie_jar = delete $cnf{cookie_jar};
     my $conn_cache = delete $cnf{conn_cache};
-- 
2.7.4

